### ISSUE TITLE
Mark request as cancelled

### ISSUE THREAD

**Issue by hazmat345** [beer-garden/beer-garden#249]
It would be nice to have something in the UI that would allow us to mark a request as CANCELLED.

Note that (and we'd have to make this explicit in the UI) this doesn't mean actually cancelling a request, but rather marking one as cancelled.

Right now we do everything possible to prevent a request from being marked as IN_PROGRESS but not existing on a queue anywhere, but sometimes it happens. In these unrecoverable cases where an admin KNOWS the request will never "finish" it would be useful to be able to mark as CANCELLED (or something) to prevent that request from contaminating the metrics forever.

**Comment by TheBurchLog** [comment]
@west270 What would be the difference between Cancelled and Error? Sounds like we should be throwing the error message instead.

**Comment by TheBurchLog** [comment]
We should add a feature to mark a Request as Abandoned. That way any System Clients waiting on it will know it won't be returned. Then if the request does complete later it can update it the record.

**Comment by scott-taubman** [comment]
This issue lines up with a recent complaint we received regarding requests getting stuck in CREATED and then there being no automatic or manual means of cleaning those up.  I suggest we take a multi-pronged approach here:

* In the [config spec](https://github.com/beer-garden/beer-garden/blob/3.13.0/src/app/beer_garden/config.py#L737) add an additional "ttl" setting called "in_progress".  This setting would be for the "Number of minutes to wait for a request in CREATED or IN_PROGRESS to complete before considering it timed out and marking it as CANCELLED".
* In the [pruner](https://github.com/beer-garden/beer-garden/blob/3.13.0/src/app/beer_garden/db/mongo/pruner.py) add logic to handle this new TTL setting.  It would search for requests in CREATED or IN_PROGRESS status that exceed the configured threshold and update their status to be "CANCELLED".
* Up for debate and probably not part of this issue regardless, on the request view page in the UI, if the request is in CREATED or IN_PROGRESS status we could display a "Cancel" button next to the "Pour it Again" button to manually mark a request as cancelled.  This could be useful if you didn't want to automatically time things out with the pruner, but wanted a mechanism to manually clean things up.
